File: src/apps/shared/src/artifact.ts
/**
 * 轻量 Artifact 引用
 */
export interface ArtifactHandle {
  id: string
  kind: string
}

/**
 * Artifact 来源信息
 */
export interface ArtifactProducer {
  type: 'agent' | 'extension' | 'input-library' | 'import'
  id: string
  runId?: string
  extensionId?: string
}

/**
 * 完整 Artifact 资源描述
 */
export interface ArtifactResource {
  id: string
  kind: string
  version?: string
  title: string
  summary?: string
  labels?: string[]
  mimeType?: string
  producer: ArtifactProducer
  fetchMode: 'object-blob' | 'api-resource' | 'external-url' | 'extension-resolver' | 'inline-json'
  descriptor: Record<string, unknown>
  capabilities?: string[]
  display?: 'inline' | 'panel'
}

/**
 * 插件 Viewer 注册信息
 */
export interface ArtifactViewerRegistration {
  extensionId: string
  supports: string[]
  openMode: 'route' | 'embedded-browser' | 'hybrid' | 'panel'
  priority: number
}

/**
 * 插件打开时的上下文
 */
export interface ArtifactContext {
  artifact: ArtifactResource
  invocation: {
    sourceSurface: 'chat' | 'input-library' | 'workspace'
    trigger: 'click' | 'command' | 'agent'
    preferredMode?: 'route' | 'embedded-browser' | 'hybrid'
  }
  relatedArtifacts?: ArtifactResource[]
  sessionContext?: unknown
}

/**
 * 系统级 kind 渲染配置（非消息字段，启动时注册）
 * inlineMode 决定 inline 状态下的渲染方式
 * viewer 指定打开时使用的 viewer（可选）
 */
export interface KindConfig {
  inlineMode: 'image' | 'iframe' | 'card-preview' | 'social-card' | 'product-card' | 'link'
  viewer?: string
}

/**
 * 内联 artifact 标记解析后的段落
 */
export type ContentSegment =
  | { type: 'text'; text: string }
  | { type: 'artifact'; id: string; kind: string; title?: string }

File: src/services/shared/artifact/artifact.go
package artifact

// Handle 轻量引用
type Handle struct {
	ID   string `json:"id"`
	Kind string `json:"kind"`
}

// Producer 来源信息
type Producer struct {
	Type     string  `json:"type"`
	ID       string  `json:"id"`
	RunID    *string `json:"runId,omitempty"`
	PluginID *string `json:"pluginId,omitempty"`
}

// Resource 完整资源描述，随消息传递
type Resource struct {
	ID           string         `json:"id"`
	Kind         string         `json:"kind"`
	Version      *string        `json:"version,omitempty"`
	Title        string         `json:"title"`
	Summary      *string        `json:"summary,omitempty"`
	Labels       []string       `json:"labels,omitempty"`
	MimeType     *string        `json:"mimeType,omitempty"`
	Producer     Producer       `json:"producer"`
	FetchMode    string         `json:"fetchMode"`
	Descriptor   map[string]any `json:"descriptor"`
	Capabilities []string       `json:"capabilities,omitempty"`
}
